components/Swap/StepsComponent.tsx

```diff
@@ -43,20 +46,51 @@ export default function Steps({ steps }) {
                                         </span>
                                     </div>
                                 </>
+                            ) : step?.status === 'failed' ? (
+                                <>
+                                    {stepIdx !== filteredSteps.length - 1 ? (
+                                        <div className="absolute top-1/2 left-4 -ml-px mt-0.5 h-[40%] w-0.5 bg-gray-300 opacity-60" aria-hidden="true" />
+                                    ) : null}
+                                    <div className={`group relative flex ${step?.description ? "items-start" : "items-center"}`}>
+                                        <span className="flex h-9 items-center" aria-hidden="true">
+                                            <XCircle className="h-8 w-8 text-red-600" aria-hidden="true" />
+                                        </span>
+                                        <span className="ml-4 flex min-w-0 flex-col">
+                                            <span className="text-sm font-medium text-gray-300">{step.name}</span>
+                                            {step?.description &&
```

Comment:
These steps have the same code, we can extract it into a Step component and reuse